src/driver/sdfmt.d

```diff
@@ -66,15 +66,18 @@ int main(string[] args) {
 		}
 
 		import format.writer;
-		auto o = chunks.write(conf);
+		const input = base.getFullPosition(context).getSource().getContent();
+		const output = chunks.write(conf);
 
 		if (inPlace) {
-			// TODO: Only write if the file changed.
-			import std.file;
-			filename.write(o);
+			import std.string : cmp;
+			if (!cmp(input, output)) {
```

Comment:
Just use ==, no?

Reply:
No actually, based on my testing at least.

Reply:
I think `a ==  b`and `a[] == b[]` use bcmp

Reply:
Didn't get down to the bottom of it, could be something to do with the utf8 canonicalisation step vs. method of outputting chunks to string